cluster/manager.go

```diff
@@ -958,9 +956,47 @@ func (c *ClusterManager) Start(
 	)
 	c.gossipVersion = types.GOSSIP_VERSION_2
 
+	dne := discovery.NodeEntry{
+		Id:            c.config.NodeId,
+		Ip:            c.selfNode.DataIp,
+		GossipVersion: types.GOSSIP_VERSION_2,
+	}
+	clusterInfo, err := c.discoverService.AddNode(dne)
```

Comment:
NodeInit is called after this routine. This one only adds self node into the discovery db and tries to connect with other node in gossip.

Yes I need to add some clean up code if NodeInit fails.